pyinfra/api/connectors/sshuserclient/client.py

```diff
@@ -18,6 +19,23 @@
 from .config import SSHConfig
 
 
+class StrictPolicy(MissingHostKeyPolicy):
+    def missing_host_key(self, client, hostname, key):
+        logger.error('No host key for {0} found in known_hosts'.format(hostname))
+        raise SSHException("StrictPolicy: No host key for {0} found in known_hosts".format(hostname))
+
+
+class AskPolicy(MissingHostKeyPolicy):
+    def missing_host_key(self, client, hostname, key):
+        logger.error('No host key for {0} found in known_hosts'.format(hostname))
+        raise SSHException("AskPolicy: not implemented!\n")
```

Comment:
This can probably just call `getpass` here the same we do for sudo passwords:

https://github.com/Fizzadar/pyinfra/blob/ff0913d6a172966760b63fe59e55dff9ea852e0d/pyinfra/api/connectors/util.py#L176

Reply:
This methode is called if paramiko does not find the ssh host key in the users `known_hosts` file, I'm not sure what exactly you mean.

Reply:
Sorry I wasn't clear - `getpass` is also the wrong function I meant `raw_input`!

```py
class AskPolicy(MissingHostKeyPolicy):
    def missing_host_key(self, client, hostname, key):
        should_continue = raw_input('No host key for {0} found in known_hosts, continue? [y/N] '.format(hostname))
        if should_continue.lower() == 'y':
                ...add key to known hosts
                return
        raise SSHException('Failed due to no valid host key for {0}'.format(hostname))
```

Reply:
I still can't find that function

Reply:
Changed, the code for adding the host key was taken from paramiko itself, but I haven't tested that it actually works.